scripts/core/ui/components/SelectUser.tsx

```diff
@@ -1,176 +1,96 @@
 /* eslint-disable react/no-multi-comp */
 import React from 'react';
-import {IPropsSelectUser, IUser, IRestApiResponse} from 'superdesk-api';
-import {gettext, getUserSearchMongoQuery} from 'core/utils';
+import {IPropsSelectUser, IUser} from 'superdesk-api';
+import {gettext} from 'core/utils';
 import {UserAvatar} from 'apps/users/components/UserAvatar';
-import {SelectWithTemplate, Spacer} from 'superdesk-ui-framework/react';
-import {httpRequestJsonLocal} from 'core/helpers/network';
+import {Spacer, TreeSelect} from 'superdesk-ui-framework/react';
 import {SuperdeskReactComponent} from 'core/SuperdeskReactComponent';
+import {store} from 'core/data';
+import ng from 'core/services/ng';
 
 interface IState {
-    selectedUser: IUser | null | 'loading';
+    selectedUser: IUser | null;
 }
 
-const itemTemplate = (props: {option: IUser}) => {
-    const user: IUser | null = props.option;
-
-    return user == null
-        ? (
-            <div>
-                {gettext('Select a user')}
-            </div>
-        )
-        : (
-            <Spacer h gap="8" noWrap justifyContent="start">
-                <div>
-                    <UserAvatar user={user} displayStatus={true} />
-                </div>
-
-                <Spacer v gap="4" noWrap>
-                    <div>{user.display_name}</div>
-                    <div style={{fontSize: '1.2rem'}}>@{user.username}</div>
-                </Spacer>
-
-            </Spacer>
-        );
-};
-
-const valueTemplateDefault = (props: {option: IUser}) => {
-    const user: IUser | null = props.option;
-
-    return user == null
-        ? (
-            <div>
-                {gettext('Select a user')}
-            </div>
-        )
-        : (
-            <Spacer h gap="8" justifyContent="start" noGrow>
-                <UserAvatar user={user} displayStatus={true} />
+export class SelectUser extends SuperdeskReactComponent<IPropsSelectUser, IState> {
+    static getDerivedStateFromProps(props: IPropsSelectUser, state: IState): Partial<IState> | null {
+        if (props.selectedUserId !== state.selectedUser?._id) {
+            const users = store.getState().entities.users;
 
-                {user.display_name}
-            </Spacer>
-        );
-};
+            return {
+                selectedUser: props.selectedUserId == null ? null : users[props.selectedUserId] ?? null,
+            };
+        }
+        return null;
+    }
 
-export class SelectUser extends SuperdeskReactComponent<IPropsSelectUser, IState> {
     constructor(props: IPropsSelectUser) {
         super(props);
+        const users = store.getState().entities.users;
 
         this.state = {
-            selectedUser: props.selectedUserId == null ? null : 'loading',
+            selectedUser: props.selectedUserId == null ? null : users[props.selectedUserId] ?? null,
```

Comment:
would it work to initialize to empty state? `getDerivedStateFromProps` should be setting the actual value